File: 643-MaximumAverageSubarrayI/643-MaximumAverageSubarrayI.java
// Last updated: 6/8/2025, 11:54:03 AM
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        double max = Double.NEGATIVE_INFINITY;
        double w_sum = 0;
        int n = nums.length;
        int start = 0;

        for(int i = 0 ; i < n ; i++){
            w_sum += nums[i];

            if(i - start + 1 == k){
                double avg = w_sum/k;
                max = Math.max(max , avg);
                w_sum -= nums[start];
                start++;
            }
        }

        return max;
    }
}

File: 643-MaximumAverageSubarrayI/643-MaximumAverageSubarrayI.py
# Last updated: 6/8/2025, 11:54:01 AM
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        n = len(nums)
        max_avg = 0
        curr_sum = 0

        for i in range(k):
            curr_sum += nums[i]
        
        max_avg = curr_sum / k

        for i in range( k , n):
            curr_sum += nums[i]
            curr_sum -= nums[i-k]

            curr_avg = curr_sum / k
            max_avg = max(curr_avg , max_avg)
        return max_avg
